src/Factory/Api/Mastodon/Status.php

```diff
@@ -155,11 +155,9 @@ public function createFromUriId(int $uriId, int $uid = 0): \Friendica\Object\Api
 			$poll = null;
 		}
 
-		$shared = Item::getShareArray($item);
-		if (!empty($shared['guid'])) {
-			$shared_item = Post::selectFirst(['uri-id', 'plink'], ['guid' => $shared['guid']]);
-
-			$shared_uri_id = $shared_item['uri-id'] ?? 0;
+		$shared = DI::contentItem()->getSharedPost($item, ['uri-id']);
```

Comment:
Please replace `DI::contentItem()` call with including the class directly at `__construct(...)` and attaching it to a new `protected $contentItem;` property.

So you can access it like
```suggestion
		$shared = $this->contentItem->getSharedPost($item, ['uri-id']);
```

Reply:
Done